contrib/icons/README.md

```diff
@@ -0,0 +1,100 @@
+directory.
+
+## Generate bitmap icons
+
+Use the bundled Makefile for that. It includes targets for rendering PNG files
+at various sizes using the `rsvg-convert` command which generates greatly
+optimised small files with correctly set alpha channel (unlike the `convert`
+command from the ImageMagick suite, which is traditionally used for this
+purpose, but it's more suited for bitmap formats).
+
+`rsvg-convert` is available in most Linux repos and via Homebrew or MacPorts
+on macOS
+
+Run `make help` to learn about the available targets.
+
+## Small icon variants
+
+The files in the `small-svg/` directory are vector icons optimized for small
+sizes; the vector format allows for easy adjustments with live preview, and
+controlling sub-pixel rendering by carefully placing the shape edges.
+
+It's recommended to use Inkscape's "icon preview" feature when modifying these
+files.
+
+
+## OS-specific instructions
+
+### Linux
+
+The base icon follows the size and colour recommendations of GNOME HIG, so you
+can use it directly, no pre-processing is needed.
+
+When packaging for Linux, copy `dosbox-staging.svg` into the distro specific
+directory (usually `/usr/share/icons/hicolor/scalable/apps/`).
+
+You will need an additional
+[.desktop](https://specifications.freedesktop.org/desktop-entry-spec/latest/)
+file to correlate the icon file to the application.
+
+Use the Makefile to generate small icons if necessary (some desktop
+environments use them).
+
+### macOS
+
+macOS does not support vector icons, so you'll need to generate an `.icns`
+file form the `dosbox-staging-macos-1024.png` source file at various smaller
+sizes:
+
+    make dosbox-staging.icns
+
+When building the app bundle, the `.icns` file must be placed in
+`Contents/SharedSupport/Resources/` and you'll also need to update the
+information in `Info.plist`.
+
+Please refer to the Apple documentation for details.
+
+### Windows
+
+Windows does not support vector icons and does not provide utilities for
+rendering/bundling bitmap icons in `.ico` format; developers are expected to
+edit icons using some image editor (e.g., the one provided by Visual Studio).
+
+For this reason, the final `.ico` file is checked in to the repo.
+
+**When changing the icon, update the generated `.ico` file only after the
+design is
+finalised.**
+
+You can rebuild the `.ico` file using `icotool` on Linux:
+
+    make dosbox-staging.ico
+
+Make sure to re-generate the icons if the following sizes when making changes
+to it:
+
+- 16x16 - Used in the window's title bar
+- 24x24 - Taskbar icon; if missing, Windows will pick the wrong icon for
+          the window's titlebar as well (it will scale down the biggest icon
+          it can find, so the result will be blurry).
+
+The icon file stored in the repo is packages into the final executable via
+Visual Studio, so make sureo t update `src/winres.rc` is you change the name
+of path of the `.ico` file.
+
```

Comment:
Awesome docs, @johnnovak :+1:

Reply:
sureo t -> sure to
is you change -> if you change